### ISSUE TITLE
Updating HighStock chart when state changes

### ISSUE THREAD

**Issue by tjugg** [highcharts/highcharts-react#48]
Hi!

How should we properly handle state updates when using HighStock? For example, my page loads with the data still being fetched, and when the data loading is completed, state updates and the HighStock chart will be given new options with the newly fetched data series. I am using this approach with HighCharts, and it works completely fine, how ever, when using HighStock, I get this error after state update and re render:
` [Show/hide message details.] TypeError: f.options is undefined[Learn More] highstock.js:508
./node_modules/highcharts/highstock.js/</</<
highstock.js:508
./node_modules/highcharts/highstock.js/</</a.fireEvent/<
highstock.js:31
forEach self-hosted:268:13 ./node_modules/highcharts/highstock.js/</</a.each
highstock.js:28
./node_modules/highcharts/highstock.js/</</a.fireEvent
highstock.js:30
setChartSize
highstock.js:261
getAxisMargins
highstock.js:256
getMargins
highstock.js:256
redraw
highstock.js:247
./node_modules/highcharts/highstock.js/</</<
highstock.js:533
./node_modules/highcharts/highstock.js/</</a.wrap/a[n]
highstock.js:20
update
highstock.js:335
componentWillReceiveProps
HighchartsReact.js:24
callComponentWillReceiveProps
react-dom.development.js:11528
updateClassInstance
react-dom.development.js:11720
updateClassComponent
react-dom.development.js:13154
beginWork
react-dom.development.js:13825
performUnitOfWork
react-dom.development.js:15864
workLoop
react-dom.development.js:15903
callCallback
react-dom.development.js:100
invokeGuardedCallbackDev
react-dom.development.js:138
invokeGuardedCallback
react-dom.development.js:187
replayUnitOfWork
react-dom.development.js:15311
renderRoot
react-dom.development.js:15963
performWorkOnRoot
react-dom.development.js:16561
performWork
react-dom.development.js:16483
performSyncWork
react-dom.development.js:16455
interactiveUpdates$1
react-dom.development.js:16720
interactiveUpdates
react-dom.development.js:2150
dispatchInteractiveEvent
react-dom.development.js:4533
dispatchInteractiveEvent self-hosted:1027:17 `

I assume that HighStock chart data updating should be handled somehow separately? Why does this approach work fine with HighChart? I have a requirement to dynamically keep adding different series to the HighStock chart, and this error is killing me :/. **I have a license also if that matters.**

EDIT: This also seems to have something to do with the Y axis. If I configure the Y axis to have for example a title, the chart crashes with the error above. If I DON'T define any config for the Y axis, the chart updates just fine. Weird. I also tried to access the chart instance via refs, and use .addSeries(), but the end result is the same.

**Comment by tjugg** [comment]
Alright, so here is an example using the ref route.

This is in the render() function 
`
<HighchartsReact
                            highcharts={HighchartsStock}
                            constructorType={'stockChart'}
                            options={chartOptions}
                            ref={'chart'}
                        />`

Here is the initial options + adding dynamically via refs: 

`const chart = this.refs.chart;

        if (chart !== undefined) {
            chart.chart.addSeries({
                name: 'OK',
                data: [1, 2, 3]
            });
            chart.chart.redraw();
        }

        if (chart === undefined) {
            return {
                title: {
                    text: 'Crash'
                },
                yAxis: {
                    title: {
                        text: 'Title'
                    }
                }
            }
        }`

Sorry for the bad formatting. So the initial chart options is what is returned in the last code snippet. That crashes for me. BUT if I remove the yAxis configuration, that chart updates just fine without crashing.

**Comment by tjugg** [comment]
https://pastebin.com/iiGjfneq this is pretty much the code.

**Comment by tjugg** [comment]
even https://github.com/highcharts/highcharts-react/blob/master/demo/demo.jsx crashes when you try to update the the state.

**Comment by scottclayton** [comment]
@tjugg and @KacperMadej 

I also have issues with the `stockChart` as well as the regular `chart` when there is `navigator.enabled = true`

Do either of you think this might essentially be the same issue as [this one in the highcharts repo about highstock navigator](https://github.com/highcharts/highcharts/issues/8196).

**Comment by tjugg** [comment]
@scottclayton I am not sure if it's the same issue, but this is definitely a bug. @KacperMadej ?

**Comment by KacperMadej** [comment]
**tl;dr** This is the same problem as the one reported in Highcharts: https://github.com/highcharts/highcharts/issues/8196 + bad demo in this repository.

**Long / full story:**
It was not possible to set `oneToOne` to `false` due to a bug in code that was fixed thank to @scottclayton - thank you very much for finding the source of the problem and submitting the PR. New version of the wrapper (1.3.2) is updating to NPM's `highcharts-react-official` as I'm typing this.

In the demo simplified structure looks like main component with 4 chart components. On first chart options update done through the main's container `setState` update check runs through it's child components and because `update` prop is by default set to true, then `shouldComponentUpdate` by default returns `true`. This causes update on other charts - stock and map charts included and because `oneToOne` for the stock chart is set to `true` the update causes the error reported in main Highcharts repository.

We are working or rewriting this wrapper to work better with new versions of React, so drastic changes are not planned until version 2 of this wrapper will be released. For now only emergency fixes are committed, so the demo is still showing the problem, but thank to the PR by @scottclayton it's avoidable by code changes.

There's no ETA for v2.0.0 for now, but it should be done within next few weeks.

@tjugg Please check with new code and `oneToOne` set to `false` for stock charts and if this doesn't help please provide more details about how the problem could be recreated - app that could be build, like this repository and it's demo, would be helpful for debugging of your problem.

**Comment by KacperMadej** [comment]
Closing due to inactivity.
